test/poolFixtures.hpp

```diff
@@ -40,6 +40,9 @@ umf_test::pool_unique_handle_t poolCreateExtUnique(poolCreateExtParams params) {
                                   &upstream_provider);
     EXPECT_EQ(ret, UMF_RESULT_SUCCESS);
     EXPECT_NE(upstream_provider, nullptr);
```

Comment:
you should probably just replace the second EXPECT_ with `ASSERT_NE(upstream_provider, nullptr);`

// unless it's not working becuase it's gtest's setup step and there's something weird going on in there ;)

Reply:
we cannot use ASSERT_* in functions that return a type other than void, because this macro returns from func

Reply:
Done